src/game_server/command.rs

```diff
@@ -41,6 +42,38 @@ impl SerializePacket for CommandOpCode {
     }
 }
 
+#[derive(SerializePacket, DeserializePacket)]
+pub struct BGRA {
+    b: u8,
+    g: u8,
+    r: u8,
+    a: u8,
+}
+
+impl BGRA {
+
+    pub fn new(b: u8, g: u8, r: u8, a: u8) -> Self {
+        BGRA { b, g, r, a }
+    }
+
+    pub fn pack(&self) -> u32 {
+        ((self.a as u32) << 24) | ((self.r as u32) << 16) | ((self.g as u32) << 8) | (self.b as u32)
+    }
```

Comment:
I think this would be good to use as the impl for the `Into<u32>` trait for more flexibility.